dvc/pkg.py

```diff
@@ -0,0 +1,120 @@
+import os
+import shutil
+
+from funcy import cached_property
+from schema import Optional
+
+from dvc.config import Config
+from dvc.cache import CacheConfig
+from dvc.exceptions import DvcException
+from dvc.utils.compat import urlparse
+
+
+class NotInstalledPkgError(DvcException):
+    def __init__(self, name):
+        super(NotInstalledPkgError, self).__init__(
+            "Package '{}' is not installed".format(name)
+        )
+
+
+class Pkg(object):
+    PARAM_NAME = "name"
+    PARAM_URL = "url"
+    PARAM_VERSION = "version"
+
+    SCHEMA = {
+        Optional(PARAM_NAME): str,
+        Optional(PARAM_URL): str,
+        Optional(PARAM_VERSION): str,
+    }
+
+    def __init__(self, pkg_dir, **kwargs):
+        self.pkg_dir = pkg_dir
+        self.url = kwargs.get(self.PARAM_URL)
+
+        name = kwargs.get(self.PARAM_NAME)
+        if name is None:
+            name = os.path.basename(self.url)
+        self.name = name
+
+        self.version = kwargs.get(self.PARAM_VERSION)
+        self.path = os.path.join(pkg_dir, self.name)
+
+    @cached_property
+    def repo(self):
+        from dvc.repo import Repo
+
+        if not self.installed:
+            raise NotInstalledPkgError(self.name)
+
+        return Repo(self.path)
+
+    @property
+    def installed(self):
+        return os.path.exists(self.path)
+
+    def install(self, cache_dir=None):
```

Comment:
It is in preparation for `dvc pkg get`, that will be able to deploy data without dvc project, so it will basically create a temp dir and will install and pull there.

Reply:
Added `dvc pkg get` that uses that.